app/Http/Controllers/Maps/MapLayerController.php

```diff
@@ -135,7 +195,79 @@ public function destroy(Map $map, MapLayer $mapLayer)
         $mapLayer->delete();
 
         return redirect()
-            ->route('maps.edit', [$map, '#tab_form-layers'])
+            ->route('maps.layers', [$map])
             ->withSuccess(__('maps/layers.delete.success', ['name' => $mapLayer->name]));
     }
+
+    /**
+     * @return \Illuminate\Http\JsonResponse
+     */
+    protected function datagridAjax($rows)
+    {
+        $html = view('layouts.datagrid._table')
+            ->with('rows', $rows)
+            ->render();
+        $deletes = view('layouts.datagrid.delete-forms')
+            ->with('models', Datagrid::deleteForms())
+            ->with('params', Datagrid::getActionParams())
+            ->render();
+
+        $data = [
+            'success' => true,
+            'html' => $html,
+            'deletes' => $deletes,
+        ];
+
+        return response()->json($data);
+    }
+
+    /**
+     * @param Request $request
+     * @param Map $map
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View|\Illuminate\Http\RedirectResponse
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function bulk(Request $request, Map $map)
+    {
+        $this->authorize('update', $map);
+        $action = $request->get('action');
+        $models = $request->get('model');
+        if (!in_array($action, $this->validBulkActions()) || empty($models)) {
+            return redirect()->back();
+        }
+
+        if ($action === 'edit') {
+            return $this->bulkBatch(route('maps.layers.bulk', ['map' => $map]), '_map-layer', $models);
+        }
+
+        $count = $this->bulkProcess($request, MapLayer::class);
+
+        return redirect()
+            ->route('maps.layers', ['map' => $map])
+            ->with('success', trans_choice('maps/layers.bulks.' . $action, $count, ['count' => $count]))
+        ;
+    }
+
+    /**
+     * Controls drag and drop reordering of map layers
+     */
+    public function reorder(ReorderLayers $request)
+    {
+        $order = 1;
+        $ids = $request->get('layer');
+        foreach ($ids as $id) {
+            $layer = MapLayer::find($id);
+            if (empty($layer)) {
```

Comment:
If I edit the ids in the html before sending, I can re-order layers from another map 🙈  needs to be fixed too on groups